src/main/java/eu/eutampieri/catacombs/ui/World.java

```diff
@@ -87,8 +85,9 @@ public void update(final long delta) {
             this.player.move(Direction.RIGHT);
         }
 
-        if (this.km.fire()) {
+        if (this.km.fire() && this.player.getWeapon().canFire()) {
             this.player.fire();
+            this.player.getWeapon().setCanFire(false);
```

Comment:
Are you sure of this? The player shoots very rarely now

Reply:
just increase weapon fire rate... i dont want the player too op

Reply:
will do